### PR TITLE
refactor: Move dark/light theme handling to a hook

### PR DESCRIPTION
**Pull request by atomrc**
## Description

This isolate the theme switching functionality to a hook and add tests to it. 
This will make it easier to update that functionality (and in a more react way of doing things)

It:
- makes the ui part of switching mode disconnected for the low layer repositories (the hook doesn't import anything from the Repositories)
- add tests for mode switching (making it easier to add functionality in the future, eg. adding system mode detection)

## Screenshots/Screencast (for UI changes)

https://github.com/wireapp/wire-webapp/assets/1090716/d437b5bc-611c-4400-a337-3dfd91836eb4

## Checklist

- [x] PR has been self reviewed by the author;
- [x] Hard-to-understand areas of the code have been commented;
- [x] If it is a core feature, unit tests have been added;

### CODE REVIEW

File: src/script/page/MainContent/panels/preferences/OptionPreferences.tsx [hunk 1]
```diff
@@ -56,7 +56,7 @@ const OptionPreferences: React.FC<OptionPreferencesProps> = ({
   } = propertiesRepository;
   const [optionAudio, setOptionAudio] = useState<AudioPreference>(settings.sound.alerts);
   const [optionReplaceInlineEmoji, setOptionReplaceInlineEmoji] = useState<boolean>(settings.emoji.replace_inline);
-  const [optionDarkMode, setOptionDarkMode] = useState<boolean>(settings.interface.theme === ThemeViewModelThemes.DARK);
+  const [optionDarkMode, setOptionDarkMode] = useState<boolean>(settings.interface.theme === 'dark');
```
Comment (atomrc) [inline src/script/page/MainContent/panels/preferences/OptionPreferences.tsx:59]: This looks like a random string, but it's actually autocompleted and typechecked by typescript. No need for `const` for that anymore

File: src/script/page/MainContent/panels/preferences/OptionPreferences.tsx [hunk 2]
```diff
@@ -103,7 +103,7 @@ const OptionPreferences: React.FC<OptionPreferencesProps> = ({
   };
 
   const saveOptionNewTheme = (useDarkMode: boolean) => {
-    const newTheme = useDarkMode ? ThemeViewModelThemes.DARK : ThemeViewModelThemes.DEFAULT;
+    const newTheme: Theme = useDarkMode ? 'dark' : 'default';
```
Comment (atomrc) [inline src/script/page/MainContent/panels/preferences/OptionPreferences.tsx:106]: This looks like a random string, but it's actually autocompleted and typechecked by typescript. No need for `const` for that anymore

File: src/script/main/app.ts [hunk 3]
```diff
@@ -346,7 +345,6 @@ export class App {
     const platformCssClass = Runtime.isDesktopApp() ? 'platform-electron' : 'platform-web';
     document.body.classList.add(osCssClass, platformCssClass);
 
-    new ThemeViewModel(this.repository.properties);
```
Comment (atomrc) [inline src/script/main/app.ts:349]: Initializing a theme should not be an `app` concern. 
The view (react component) should be the layer that is responsible for initializing those kind of things

File: src/script/components/AppContainer/hooks/useTheme.test.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import {amplify} from 'amplify';
+
+import {WebAppEvents} from '@wireapp/webapp-events';
+
+import {Theme, useTheme} from './useTheme';
+
+describe('useTheme', () => {
+  it.each([
+    ['dark', 'theme-dark'],
+    ['default', 'theme-default'],
```
Comment (PatrykBuniX) [inline src/script/components/AppContainer/hooks/useTheme.test.ts:30]: question: Are we actually using those?

Comment (atomrc) [reply src/script/components/AppContainer/hooks/useTheme.test.ts:30]: I'm stupid I think 🤦 😆

File: src/script/page/MainContent/panels/preferences/OptionPreferences.tsx (hunk 1 above)
Comment (PatrykBuniX) [reply src/script/page/MainContent/panels/preferences/OptionPreferences.tsx:59]: praise: I find literal strings even more readable than enums/consts (if they are strongly typed) 😎

File: src/script/components/AppContainer/hooks/useTheme.test.ts (hunk 4 above)
Comment (atomrc) [reply src/script/components/AppContainer/hooks/useTheme.test.ts:30]: fixed 👼 
https://github.com/wireapp/wire-webapp/pull/15709/commits/51eabc3448330f51d1fc80df0a4b0f56ba37d31f

File: src/script/components/AppContainer/hooks/useTheme.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+import type {WebappProperties} from '@wireapp/api-client/lib/user/data';
+import {amplify} from 'amplify';
+
+import {WebAppEvents} from '@wireapp/webapp-events';
+
+const THEMES_CLASS_PREFIX = 'theme-';
+
+export type Theme = WebappProperties['settings']['interface']['theme'];
+
+const listenenEvents = [WebAppEvents.PROPERTIES.UPDATE.INTERFACE.THEME, WebAppEvents.PROPERTIES.UPDATED];
```
Comment (przemvs) [inline src/script/components/AppContainer/hooks/useTheme.ts]: listenenEvents? :D 
eventsListener | listenedEvents sounds better :)

**Review by przemvs — APPROVED** [review]

**Review by PatrykBuniX — APPROVED** [review]

**Pull request merged**